src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -108,7 +108,8 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
 
                     Debug.Assert(expectedType.IsCollection());
                     string expectedTypeFullName = expectedType.Definition.AsElementType().FullTypeName();
-                    if (expectedTypeFullName.Equals("Edm.String", StringComparison.Ordinal))
+
+                    if (expectedTypeFullName.Equals("Edm.String", StringComparison.Ordinal) || expectedTypeFullName.Equals("Edm.Untyped", StringComparison.Ordinal))

```

Comment:
Can the expectedFullTypeName be null at this point?

Reply:
No it can't.